rust/private/clippy.bzl

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(
+    "@io_bazel_rules_rust//rust:private/rustc.bzl",
+    "CrateInfo",
+    "collect_deps",
+    "collect_inputs",
+    "construct_arguments",
+    "construct_compile_command",
+)
+load(
+    "@io_bazel_rules_rust//rust:private/rust.bzl",
+    "crate_root_src",
+    "get_edition"
+)
+load("@io_bazel_rules_rust//rust:private/utils.bzl", "find_toolchain")
+
+_rust_extensions = [
+    "rs",
+]
+
+def _is_rust_target(srcs):
+    return any([src.extension in _rust_extensions for src in srcs])
+
+def _all_sources(target, rule):
+    srcs = []
+    if "srcs" in dir(rule.attr):
+        srcs += [f for src in rule.attr.srcs for f in src.files.to_list()]
+    if "hdrs" in dir(rule.attr):
+        srcs += [f for src in rule.attr.hdrs for f in src.files.to_list()]
+    return srcs
+
+def _rust_sources(target, rule):
+    srcs = _all_sources(target, rule)
```

Comment:
Inline the all sources then the line below can be included in the previous iteration

Reply:
I've inlined the all sources line, let me know if this is what you meant.

Reply:
I don't see the change (might be an issue with Github). What I meant is to inline `_all_sources` method in `_rust_sources` and then the line 47 can be merged with lines 40 and 42 (just puting the if in those lines)

Reply:
Done (squashed the two methods into one)